src/lib/hash.ts

```diff
@@ -35,14 +35,16 @@ class Sponge {
 
 const Poseidon = {
   hash(input: Field[]) {
-    let isChecked = inCheckedComputation();
+    let isChecked = !input.every((x) => x.isConstant());
```

Comment:
this is a better (more general) way of determining whether to use a "checked" implementation, enables us to clean up code below